svg-injector.js

```diff
@@ -180,14 +172,32 @@
     }
   };
 
+  // Returns SVGSVGElement or undefined
+  var parseSvg = function(svgString){
+    if (DOMParser && (DOMParser instanceof Function)) {
+      var xmlDoc;
+      try {
+        var parser = new DOMParser();
+        xmlDoc = parser.parseFromString(svgString, 'text/xml');
+      }
+      catch (e) {
+        return;
+      }
+
+      if (xmlDoc && !xmlDoc.getElementsByTagName('parsererror').length)
+        return xmlDoc;
+    }
+  }
+
   // Inject a single element
   var injectElement = function (el, evalScripts, pngFallback, callback) {
 
     // Grab the src or data-src attribute
     var imgUrl = el.getAttribute('data-src') || el.getAttribute('src');
 
-    // We can only inject SVG
-    if (!(/\.svg/i).test(imgUrl)) {
+    // We can only load SVG, test for warm cache, and check 
+    // extension if not already loaded
+    if (!(svgCache[imgUrl] || (!hasSvgSupport && pngCache[imgUrl])) && !(/\.svg/i).test(imgUrl)) {
```

Comment:
added conditions are not related to non-svg extension, like the returned error states